library/ssl_misc.h

```diff
@@ -2482,7 +2482,7 @@ static inline int mbedtls_ssl_tls12_sig_alg_is_supported(
             break;
 #endif
 
-#if defined(MBEDTLS_PK_CAN_ECDSA_SOME)
+#if defined(MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED)
```

Comment:
I don't think that's correct, we can use ECDSA for client auth also with other key exchanges, I think `MBEDTLS_KEY_EXCHANGE_ECDSA_CERT_REQ_ALLOWED_ENABLED` is what we want here.

Reply:
I apologize for this as well as for the following errors. When replacing previous symbols with `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` I thought that those part of the TLS's code were not related to client authentication, but rather being limited to the key exchange itself.
Obviously that was not correct as add the new components you suggested to `all.sh` I had `ssl-opt.sh` start failing.
I will go through all of those changes and fix them.

Reply:
No worries, this part of the protocol is quite complex. I wasn't 100% sure myself about the things I pointed out, which is why I requested tests, that way we know for sure :)